src/lightgbm/src/main/scala/TrainUtils.scala

```diff
@@ -158,7 +158,7 @@ private object TrainUtils extends Serializable {
   }
 
   def translate(labelColumn: String, featuresColumn: String, weightColumn: Option[String],
-                validationData: Option[Array[Row]], log: Logger,
+                validationData: Option[Broadcast[Array[Row]]], log: Logger,
```

Comment:
my reasoning was that passing it through a function isn't expensive and helps indicate to the person reading the code that this variable was broadcast.  Also, I wanted to be extra careful that we don't get the value too early, eg:
https://stackoverflow.com/questions/50418059/spark-which-is-the-proper-way-to-use-a-broadcast-variable?noredirect=1&lq=1
"The argument should be of type Broadcast[_] so don't use rdd1Broadcast.value. If parameter is passed by value, it will be evaluated and substituted locally, and broadcast will not be used."